### PR TITLE
feat: add unit tests for Helm chart

### PR DESCRIPTION
**Pull request by jawnsy**
Very rough first pass, mostly a proof-of-concept. The tests aren't complete, and not part of CI yet, but the goal is to show what writing them could look like.

### CODE REVIEW

File: tests/values.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var _ = fmt.Stringer(CoderValues{})
+
+// CoderValues is a typed Go representation of Coder's values file,
+// suitable for writing tests. This provides code completion for Go
+// tests.
+type CoderValues struct {
```
Comment (johnstcn) [inline tests/values.go:30]: Keeping this in sync manually is going to be an issue down the line.
Could we generate this automatically from `values.yaml`? 

Aside: _is_ there any decent tool for doing this directly? I found [yaml2go](https://github.com/PrasadG193/yaml2go) but it appears to not be maintained. We could also go the route of YAML -> JSON -> Go, for which there is a [slightly more up to date tool](https://github.com/m-zajac/json2go). If all else fails, I suppose there's Swagger.

**Review by johnstcn — APPROVED** [review]
I really like the shape of this. Being able to just take a `values.yaml`, render the Helm chart, and make assertions about the output is really powerful!

File: tests/values.go (hunk 1 above)
Comment (jawnsy) [reply tests/values.go:30]: I discussed this with @Emyrk, there's a way you can define a JSON Schema to validate the values, and we could generate both the values.yaml and these structs from that. But in the interest of an initial implementation, I left taht out for now :)

Comment (johnstcn) [reply tests/values.go:30]: Nice :-) I think this will really help!

**Pull request merged**